apollo-federation/src/operation/mod.rs

```diff
@@ -1330,14 +1330,32 @@ mod field_selection {
         }
     }
 
+    // SiblingTypename indicates how the sibling __typename field should be restored.
+    // PORT_NOTE: The JS version used the empty string to indicate unaliased sibling typenames.
+    // Here we use an enum to make the distinction explicit.
+    #[derive(Debug, Clone)]
+    pub(crate) enum SiblingTypename {
```

Comment:
if we are using custom enum to represent the sibling typename -> maybe we should drop the `Option`? so we would have `None/Unaliased/Aliased`?

Reply:
Technically we could still use just `Option<Name>` without enum -> ie just initialize it with `__typename`

Reply:
Using a special name as non-alias would throw someone off at some point. So, I didn't want to do that.

I did consider `None/Unaliased/Aliased`, but I liked the added `Option` after all. First, the sibling typename handling logic for Unaliased/Aliased will take the same branch anyways to add `__typename`. The added `fn alias()` getter function, which returns `Option<Name>`, makes no further case-split necessary. After all, that's just a subtle difference. So, I took the path of least resistance, not changing all the existing `None` cases :)